scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/GitRepositoryService.java

```diff
@@ -0,0 +1,81 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.diging.scriptoocloud.core.data.GitRepositoryRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.InvalidGitUrlException;
+import edu.asu.diging.scriptoocloud.core.model.GitRepository;
+import edu.asu.diging.scriptoocloud.core.model.impl.GitRepositoryImpl;
+import edu.asu.diging.scriptoocloud.core.service.DeleteFilesService;
+import edu.asu.diging.scriptoocloud.core.service.GitRepositoryManager;
+import edu.asu.diging.scriptoocloud.core.service.JgitService;
+import edu.asu.diging.scriptoocloud.core.service.UrlFormatterUtility;
+
+@Service
+@Transactional
+@PropertySource("classpath:config.properties")
+public class GitRepositoryService implements GitRepositoryManager{
+
+    @Autowired
+    private DeleteFilesService deleteFilesService;
+    
+    @Autowired
+    private UrlFormatterUtility urlFormatter;
+    
+    @Autowired 
+    private GitRepositoryRepository gitRepositoryJpa;
+    
+    @Autowired
+    private JgitService jGitService;
+
+    @Value("${git.repositories.path}")
+    public String path;
+
+    @Override
+    public void cloneRepository(String gitUrl, String requester) throws InvalidGitUrlException, MalformedURLException{
+        String folderName = urlFormatter.urlToFolderName(gitUrl);
+
+        ZonedDateTime creationDate = ZonedDateTime.now();                                   
+      
+        GitRepositoryImpl gitRepository = new GitRepositoryImpl();
+        gitRepository.setUrl(gitUrl);
+        gitRepository.setRequester(requester);
+        gitRepository.setCreationDate(creationDate);
+        gitRepository.setFolderName(folderName);
+        
+        GitRepository repositoryEntity = gitRepositoryJpa.findByUrl(gitUrl);
+        
+        if(repositoryEntity != null){
+            deleteRepository(repositoryEntity.getId());
```

Comment:
hm, I wouldn't delete the repository object here, but use the existing one and update it. You'll probably have dependencies to the repository objects later (e.g. from the projects that use a repo) and then you either wouldn't be able to delete the repo here without removing the relationship from projects to repo, or you would have links from projects to repo objects that don't exist anymore.